tests/homework1/test_task05.py

```diff
@@ -15,3 +15,7 @@ def test_find_maximal_subarray_sum(nums: list, k: int, expected_result: int):
     our_result = find_maximal_subarray_sum(nums, k)
 
     assert our_result == expected_result
+
+def test_find_maximal_subarray_sum_exception():
```

Comment:
Also add test cases with negative `k` and empty array to check all of the conditions.